src/http/peer.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 The Forgotten Server Authors. All rights reserved.
+// Use of this source code is governed by the GPL-2.0 License that can be found in the LICENSE file.
+
+#include "peer.h"
+#include "server.h"
+#include "router.h"
+#include "responder.h"
+#include "../tasks.h"
+
+namespace Http
+{
+
+using Minutes = boost::posix_time::minutes;
+
+/** \brief Determines the time after which a connection is closed due to inactivity
+ */
+const static Minutes TIMER_TIMEOUT{30};
+
+Peer::Peer(Server& server, Router& router, PeerId peerId) :
+	server(server),
+	router(router),
+	socket(server.getIoService()),
+	timer(server.getIoService()),
+	strand(server.getIoService()),
+	peerId(peerId)
+{
+	//
+}
+
+void Peer::onAccept()
+{
+	auto sharedThis = shared_from_this();
+	g_dispatcher.addTask(createTask([sharedThis, this]() {
```

Comment:
```suggestion
	g_dispatcher.addTask(createTask([sharedThis = shared_from_this(), this]() {
```

And remove the line above